simhash/__init__.py

```diff
@@ -100,7 +104,20 @@ def build_by_text(self, content):
         features = self._tokenize(content)
         features = {k:sum(1 for _ in g) for k, g in groupby(sorted(features))}
         return self.build_by_features(features)
+        
+    def concatenate(self, objs):
+        if isinstance(objs, collections.Iterable):
+            digests = [int_to_bytes(self.value, self.f_bytes)]
+            digests.extend([int_to_bytes(obj.value, obj.f_bytes) for obj in objs])
+            new_value = bytes_to_int(b''.join(digests))
+            new_length = self.f
+            for obj in objs:
+                new_length += obj.f
```

Comment:
Hi @orapic 
Sorry, I think I misunderstood the `concatenate` function previously.
I see you are changing the `f` value, but I thought we were not changing it.
May I ask a real use case for this `concatenate` function?

Reply:
Hi @1e0ng ,
The `f` value changes not for the previous simhashes but for the new one.
So if we have two simhashes of 16 and we concatenate them we have a new simhash of 32.
`0xaaaa` concatenated with `0xbbbb` gives `0xaaaabbbb`. The new `f` needs to be the sum of the simhashes' length used.

Im currently using it not really for text representation but to represent other things. Lets say you have an object represented as a class and it has different parts to it you really want to differentiate. For example, a house. Houses can have its exterior color, interior color, empty or not, is being sold, etc. So you can represent the house as a 32 bit simhash where you have (for example):

- 8 bits for the interior color
- 8 bits for the exterior color
- 8 bits to represent if the house is empty or not
- 8 bits to represent if its being sold or not

Lets say now that we want to search all the houses that hame similar properties to the one we want, we would search for simhashes with closer hamming distance.
This is just a quick example, Im just currently working on something a little bit more complex and added the feature because it was useful for me.

If you think its not suitable because few people will use it, I would understand, we can close the pull request.

Reply:
Hi,  thanks for sharing your use case. It's a very interesting one. At the moment, I'm not sure the one you concatenated with 2 smihash can still be called with a simhash. On the other hand, even the building method is different, I'm also not sure that it is not a simhash. I'll check the definition of simhash in the thesis and revert back.

Reply:
Hi, after some thoughts, I think it might be better we create a new class `MultiSimhash` to handle this scenario, and this new class can have a constructor function that accepts a list of `Simhash` objects. Would that make sense to you?

Reply:
Hi again,
Sure that would be a viable option. I'll work on it as soon as I'm able to. I'm not used to but I guess this approach would require this new class to inherit from `Simhash`, right?